proto/provenance/metadata/v1/query.proto

```diff
@@ -21,96 +21,198 @@ service Query {
   }
 
-  // Scope returns a specific scope by id
+  // ---- Primary Data Queries -----
+
+  // Scope searches for a scope.
+  //
+  // The scope id, if provided, must either be scope uuid, e.g. 91978ba2-5f35-459a-86a7-feca1b0512e0 or a scope address,
+  // e.g. scope1qzge0zaztu65tx5x5llv5xc9ztsqxlkwel. The session addr, if provided, must be a bech32 session address,
+  // e.g. session1qxge0zaztu65tx5x5llv5xc9zts9sqlch3sxwn44j50jzgt8rshvqyfrjcr. The record_addr, if provided, must be a
+  // bech32 record address, e.g. record1q2ge0zaztu65tx5x5llv5xc9ztsw42dq2jdvmdazuwzcaddhh8gmu3mcze3. Case doesn't
+  // matter.
+  //
+  // Providing a session addr or record addr does not limit the sessions and records returned.
+  // Those parameters are only used to find the scope.
+  //
+  // * If a scope id is provided, the other input is ignored and that scope is returned.
+  // * Otherwise, if a session addr is provided, the record addr is ignored and the scope containing that session is
+  // returned.
+  // * Lastly, if a record addr is provided, the scope containing that record is returned.
+  //
+  // By default, sessions and records are not included.
+  // Set include_sessions and/or include_records to true to include sessions and/or records.
   rpc Scope(ScopeRequest) returns (ScopeResponse) {
-    option (google.api.http).get = "/provenance/metadata/v1/scope/{scope_uuid}";
+    option (google.api.http) = {
+      get: "/provenance/metadata/v1/scope/{scope_id}"
+      additional_bindings: [
+        {get: "/provenance/metadata/v1/session/{session_addr}/scope"},
+        {get: "/provenance/metadata/v1/record/{record_addr}/scope"}
+      ]
+    };
   }
 
-  // SessionContextByUUID returns a specific session context within a scope (or all sessions)
-  rpc SessionContextByUUID(SessionContextByUUIDRequest) returns (SessionContextByUUIDResponse) {
-    option (google.api.http).get = "/provenance/metadata/v1/scope/uuid/{scope_uuid}/session/{session_uuid}";
+  // ScopesAll retrieves all scopes.
+  rpc ScopesAll(ScopesAllRequest) returns (ScopesAllResponse) {
+    option (google.api.http).get = "/provenance/metadata/v1/scopes/all";
   }
 
-  // SessionContextByID returns a specific session context within a scope (or all sessions)
-  rpc SessionContextByID(SessionContextByIDRequest) returns (SessionContextByIDResponse) {
-    option (google.api.http).get = "/provenance/metadata/v1/scope/id/{scope_id}/session/{session_id}";
+  // Sessions searches for sessions.
+  //
+  // The scope id can either be scope uuid, e.g. 91978ba2-5f35-459a-86a7-feca1b0512e0 or a scope address, e.g.
+  // scope1qzge0zaztu65tx5x5llv5xc9ztsqxlkwel. Similarly, the session_id can either be a uuid or session address, e.g.
+  // session1qxge0zaztu65tx5x5llv5xc9zts9sqlch3sxwn44j50jzgt8rshvqyfrjcr. Case doesn't matter.
+  //
+  // * If only a session id is provided, and it is an address, then the scope id will be ignored and the session id used
+  // to find a single session.
```

Comment:
If the scope id within the session id doesn't match the included scope-id this seems more like an error condition (invalid request/id mis-match) than an ignore case.